reusable_workflows/repo_policies/bot_checks/check_bot_approved_files.py

```diff
@@ -22,8 +22,13 @@ def get_changed_files(merge_base_sha: str, branch_head_sha: str) -> list[str]:
     """
     commit_range = f"{merge_base_sha}..{branch_head_sha}"
     result = subprocess.run(
-        ["git", "diff", "--name-only", commit_range], stdout=subprocess.PIPE, text=True
+        ["git", "diff", "--name-only", commit_range],
+        stdout=subprocess.DEVNULL,
+        stderr=subprocess.DEVNULL,
+        text=True,
     )
+    if result.returncode != 0:
+        raise RuntimeError(f"git diff failed: {result.stderr}")
```

Comment:
Wouldn't `result.stderr` be `None` in this case?

I believe you need to set `capture_output=True` and no set `stderr=subprocess.DEVNULL`.

Reply:
Thanks, admittedly I still don't know what best practice is for calling `subprocess.run`. Is this all that's needed?

Reply:
I don't know if there's a best practise. It just depends on what behaviour you want. I think just read [the docs](https://docs.python.org/3/library/subprocess.html#subprocess.run) carefully and convince yourself that it's correct. Then do test it of course.